modules/internal/SharedObject.chpl

```diff
@@ -251,8 +247,8 @@ module SharedObject {
        that refers to the same class instance as `src`.
        These will share responsibility for managing the instance.
      */
-    proc init=(type ThisType, pragma "nil from arg" const ref src:_shared(?)) {
-      this.chpl_t = ThisType.chpl_t;
+    proc init=(pragma "nil from arg" const ref src:_shared(?)) {
+      this.chpl_t = this.type.chpl_t;
```

Comment:
This seems strange. If `this.type.chpl_t` is always known, why do we need to set it at all? Can you simply remove this line now?

Reply:
Not yet, the old initializer rules still apply to init= so every field needs to be initialized explicitly (except for those with default values). I think we can remove the need for this line in the future.